torchrl/data/tensor_specs.py

```diff
@@ -2582,7 +2582,10 @@ def __init__(self, *args, shape=None, device=None, **kwargs):
                     item = CompositeSpec(item, shape=shape)
                 if item is not None:
                     if self._device is None:
-                        self._device = item.device
+                        try:
+                            self._device = item.device
+                        except RuntimeError:
+                            self._device = item._device
```

Comment:
I added this because otherwise it would throw an error when trying to clone a CompositeSpec with no device